File: 刷题/golang/1779. 找到最近的有相同 X 或 Y 坐标的点.go
func nearestValidPoint(x int, y int, points [][]int) (ans int) {
    ans=-1
    minD:=math.MaxInt32
    for i,point:=range points{
        if(point[0]==x||point[1]==y){
            d:=abs(x-point[0])+abs(y-point[1])
            if(d<minD){
                minD=d
                ans=i
            }
        }
    }
    return 
}
func abs(x int) int{
    if x<0 {
        return -x
    }else{
        return x
    }
}

File: 刷题/cpp/1779. 找到最近的有相同 X 或 Y 坐标的点.cpp
class Solution {
public:
    int distance(int x,int y,int xx,int yy){
        return abs(x-xx)+abs(y-yy);
    }
    int nearestValidPoint(int x, int y, vector<vector<int>>& points) {
        int ans=-1;
        int minD=0x3fffffff;
        for(int i=0;i<points.size();i++){
            if(points[i][0]==x||points[i][1]==y){
                int d=distance(x,y,points[i][0],points[i][1]);
                if(d<minD){
                    minD=d;
                    ans=i;
                }
            }
        }
        return ans;
    }
};